esp/services/ws_store/espstorelib/espstorebase.hpp

```diff
@@ -0,0 +1,111 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma warning (disable : 4786)
+
+#ifndef _ESPSTOREBASE_HPP__
+#define _ESPSTOREBASE_HPP__
+
+#ifdef ESPSTORE_EXPORTS
+    #define ESPSTORE_API DECL_EXPORT
+#else
+    #define ESPSTORE_API DECL_IMPORT
+#endif
+
+#include "jiface.hpp"
+
+#include "dautils.hpp"
+#include "dasds.hpp"
+#include "SecureUser.hpp"
+
+#define SDS_LOCK_TIMEOUT_KVSTORE (30*1000)
+
+interface IEspStore : extends IInterface
+{
+    virtual bool init(const char * name, const char * type, IPropertyTree * cfg) = 0;
+
+    virtual bool set(const char * storename, const char * thenamespace, const char * key, const char * value, ISecUser * owner, bool global) = 0;
+    virtual StringArray fetchKeySet(const char * storename, const char * ns, ISecUser * user, bool global) = 0;
```

Comment:
Recommend having the user pass in a reference to a StringArray that it creates, instead of returning a reference to the service's StringArray